File: packages/backend/app/schemas/room.py
from pydantic import BaseModel


class RoomAnalysisResponse(BaseModel):
    room_type: str
    walls: list[dict]
    windows: list[dict]
    doors: list[dict]
    floor: dict
    ceiling: dict
    furniture: list[dict]
    lighting: dict
    estimated_dimensions: dict
    style_assessment: str


class RoomCreateRequest(BaseModel):
    project_id: int
    name: str
    room_type: str | None = None


class RoomResponse(BaseModel):
    id: int
    project_id: int
    name: str
    room_type: str | None
    dimensions: dict | None
    analysis_data: dict | None
    created_at: str

    class Config:
        from_attributes = True

File: packages/shared/src/types/room.ts
export interface RoomAnalysis {
  roomType: string;
  walls: WallInfo[];
  windows: WindowInfo[];
  doors: DoorInfo[];
  floor: FloorInfo;
  ceiling: CeilingInfo;
  furniture: FurnitureItem[];
  lighting: LightingInfo;
  estimatedDimensions: RoomDimensions;
  styleAssessment: string;
}

export interface WallInfo {
  id: number;
  position: string; // "noord", "oost", "zuid", "west"
  color: string; // hex
  condition: string;
  areaM2: number | null;
}

export interface WindowInfo {
  id: number;
  type: string; // "draairaam", "schuifraam", "dakraam"
  wallPosition: string;
  estimatedSize: string;
}

export interface DoorInfo {
  id: number;
  type: string; // "binnendeur", "buitendeur", "schuifdeur"
  wallPosition: string;
}

export interface FloorInfo {
  type: string; // "laminaat", "tegels", "hout", "beton"
  condition: string;
  color: string;
}

export interface CeilingInfo {
  type: string; // "gestuukt", "plafondplaten", "houten balken"
  color: string;
  height: number | null;
}

export interface FurnitureItem {
  name: string;
  type: string;
  estimatedSize: string;
  position: string;
  style: string;
}

export interface LightingInfo {
  natural: string;
  artificial: string[];
  overallBrightness: string;
}

export interface RoomDimensions {
  widthM: number | null;
  lengthM: number | null;
  heightM: number | null;
}

export interface Room {
  id: number;
  projectId: number;
  name: string;
  originalImage: string;
  roomType: string | null;
  dimensions: RoomDimensions | null;
  analysisData: RoomAnalysis | null;
  createdAt: string;
}
